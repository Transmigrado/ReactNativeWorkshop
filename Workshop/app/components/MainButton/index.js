import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'

const MainButton = ({ styleButton = 'STYLE_1', label}) => {

    const getStyle = () => {
        if(styleButton == 'STYLE_2'){
            return styles.style2
        }

        return {}
    }

    const getStyleLabel = () => {
        if(styleButton == 'STYLE_2'){
            return styles.label2
        }

        return {}
    }

    return (
        <TouchableOpacity style={[styles.container, getStyle()]}>
            <Text style={[styles.label, getStyleLabel()]}>{label}</Text>
        </TouchableOpacity>
    )
}

MainButton.propTypes = {
    styleButton: PropTypes.oneOf(['STYLE_1', 'STYLE_2']),
    label: PropTypes.string
}

const styles = StyleSheet.create({
    container:{
        backgroundColor: '#FFFFFF',
        paddingHorizontal: 32,
        paddingVertical: 15,
        alignItems:'center',
        justifyContent: 'center',
        borderRadius: 4
    },
    style2:{
        backgroundColor:'#3F5498',
        borderColor:'#6F7EB1',
        borderWidth: 1
    },
    label:{
        fontWeight:'bold',
        fontSize: 11,
        color:'#01305E'
    },
    label2:{
       color:'#FFFFFF' 
    }
})

export default MainButton