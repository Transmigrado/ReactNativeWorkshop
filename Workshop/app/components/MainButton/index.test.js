
import MainButton from './index'
import React from 'react'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

configure({adapter: new Adapter()})

describe('test snapshot', () => {
    it('renders correctly for styleButton is STYLE_1', () => {
        const tree = renderer
                        .create(<MainButton styleButton="STYLE_1" label="Hola Mundo" />)
                        .toJSON()
    
        expect(tree).toMatchSnapshot()
    })
    
    it('renders correctly for styleButton is undefined', () => {
        const tree = renderer
                        .create(<MainButton label="Hola Mundo" />)
                        .toJSON()
    
        expect(tree).toMatchSnapshot()
    })
    
    it('renders correctly for styleButton is STYLE_2', () => {
        const tree = renderer
                        .create(<MainButton styleButton="STYLE_2" label="Hola Mundo" />)
                        .toJSON()
    
        expect(tree).toMatchSnapshot()
    })
})


