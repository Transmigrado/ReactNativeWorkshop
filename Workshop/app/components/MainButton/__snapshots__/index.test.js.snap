// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot renders correctly for styleButton is STYLE_1 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 4,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#01305E",
          "fontSize": 11,
          "fontWeight": "bold",
        },
        Object {},
      ]
    }
  >
    Hola Mundo
  </Text>
</View>
`;

exports[`test snapshot renders correctly for styleButton is STYLE_2 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3F5498",
      "borderColor": "#6F7EB1",
      "borderRadius": 4,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#01305E",
          "fontSize": 11,
          "fontWeight": "bold",
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Hola Mundo
  </Text>
</View>
`;

exports[`test snapshot renders correctly for styleButton is undefined 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 4,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#01305E",
          "fontSize": 11,
          "fontWeight": "bold",
        },
        Object {},
      ]
    }
  >
    Hola Mundo
  </Text>
</View>
`;
